File: server-java/src/main/java/com/jixiao2/server/evaluation/EvaluationOpsController.java
package com.jixiao2.server.evaluation;

import com.jixiao2.server.web.CurrentUser;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/admin/evaluation")
public class EvaluationOpsController {

  private final EvaluationService evaluationService;

  public EvaluationOpsController(EvaluationService evaluationService) {
    this.evaluationService = evaluationService;
  }

  @GetMapping("/performance-periods")
  public Map<String, Object> performancePeriods(@CurrentUser String userId) {
    return evaluationService.listPerformancePeriods(userId);
  }

  @GetMapping("/leaderboard")
  public Map<String, Object> leaderboard(
      @CurrentUser String userId,
      @RequestParam String scope,
      @RequestParam String key,
      @RequestParam(name = "departmentIds", required = false) String departmentIdsRaw) {
    List<String> departmentIds = null;
    if (departmentIdsRaw != null && !departmentIdsRaw.trim().isEmpty()) {
      departmentIds = new ArrayList<String>();
      for (String part : departmentIdsRaw.split(",")) {
        String trimmed = part.trim();
        if (!trimmed.isEmpty()) {
          departmentIds.add(trimmed);
        }
      }
    }
    return evaluationService.getLeaderboard(userId, scope, key, departmentIds);
  }

  @GetMapping("/awards")
  public Map<String, Object> listAwards(@CurrentUser String userId, @RequestParam String periodId) {
    return evaluationService.listPeriodAwards(userId, periodId);
  }

  @PostMapping("/awards")
  public Map<String, Object> createAward(@CurrentUser String userId, @RequestBody Map<String, Object> body) {
    return evaluationService.createPeriodAward(userId, body);
  }

  @DeleteMapping("/awards/{id}")
  public Map<String, Boolean> removeAward(@CurrentUser String userId, @PathVariable("id") String id) {
    evaluationService.removePeriodAward(userId, id);
    Map<String, Boolean> out = new LinkedHashMap<String, Boolean>();
    out.put("success", Boolean.TRUE);
    return out;
  }
}

File: legacy/nest-server/modules/evaluation/evaluation-ops.controller.ts
import {
  Controller,
  Get,
  Post,
  Delete,
  Body,
  Param,
  Query,
  Req,
  UseGuards,
} from '@nestjs/common';
import type { Request } from 'express';
import { SessionUserGuard } from '../../guards/session-user.guard';
import { EvaluationService } from './evaluation.service';
import type { CreatePeriodAwardRequest } from '@shared/api.interface';

@Controller('api/admin/evaluation')
@UseGuards(SessionUserGuard)
export class EvaluationOpsController {
  constructor(private readonly evaluationService: EvaluationService) {}

  @Get('performance-periods')
  performancePeriods(@Req() req: Request) {
    return this.evaluationService.listPerformancePeriods(req.userContext!.userId!);
  }

  @Get('leaderboard')
  leaderboard(
    @Req() req: Request,
    @Query('scope') scope: 'month' | 'quarter',
    @Query('key') key: string,
    @Query('departmentIds') departmentIdsRaw?: string,
  ) {
    const departmentIds = departmentIdsRaw
      ? departmentIdsRaw
          .split(',')
          .map((s) => s.trim())
          .filter(Boolean)
      : undefined;
    return this.evaluationService.getLeaderboard(req.userContext!.userId!, {
      scope,
      key,
      departmentIds,
    });
  }

  @Get('awards')
  listAwards(@Req() req: Request, @Query('periodId') periodId: string) {
    return this.evaluationService.listPeriodAwards(req.userContext!.userId!, periodId);
  }

  @Post('awards')
  createAward(@Req() req: Request, @Body() body: CreatePeriodAwardRequest) {
    return this.evaluationService.createPeriodAward(req.userContext!.userId!, body);
  }

  @Delete('awards/:id')
  async removeAward(@Req() req: Request, @Param('id') id: string) {
    await this.evaluationService.removePeriodAward(req.userContext!.userId!, id);
    return { success: true };
  }
}
